### PR TITLE
[SAIC-711] Create routers in its tenant

### PR DESCRIPTION
**Pull request by SVilgelm**

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:291: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 25)
cloudferrylib/os/network/neutron.py:805: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:909: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:934: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:965: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1026: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by archyufa** [comment]
Didin't work in staging. Still failing with same error. 
Though additional re-run migrates the image

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:291: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 25)
cloudferrylib/os/network/neutron.py:804: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:908: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:933: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:964: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1025: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:291: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 25)
cloudferrylib/os/network/neutron.py:805: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:909: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:934: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:965: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1026: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:805: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:785: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1790/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:785: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

File: cloudferrylib/os/network/neutron.py [hunk 1]
```diff
@@ -106,14 +106,11 @@ def read_info(self, **kwargs):
...
+        routers = self.get_routers()
         subnet_ids = {sn['id'] for sn in subnets}
-        for router in self.get_routers():
-            router_subnet_ids = subnet_ids & set(router['subnet_ids'])
-            if router_subnet_ids:
-                router['subnet_ids'] = list(router_subnet_ids)
-                routers.append(router)
-                LOG.debug('Got router: %s', router['id'])
+        for router in routers:
+            router['subnet_ids'] = [sn_id for sn_id in router['subnet_ids']
```
Comment (Settis) [inline cloudferrylib/os/network/neutron.py]: Why are you filter router subnets?
For example we have on SRC:
```
Tenant 1 with public_notShared_network_t1 <=> router_t1 <=> private_notShared_network_t1
Tenant 2 with public_notShared_network_t2 <=> router_t2 <=> private_notShared_network_t2
```
Let's do migration by 2 steps:
1. filter by `Tenant 1` and migrate. DST:
```
public_notShared_network_t1 <=> router_t1 <=> private_notShared_network_t1
public_notShared_network_t2 <=> router_t2 
```
2. filter by `Tenant 2` and migrate. DST:
```
public_notShared_network_t1 <=> router_t1 <=> private_notShared_network_t1
public_notShared_network_t2 <=> router_t2 <=> private_notShared_network_t2
                            <=> router_t2
```
You will have 2 rotuers `router_t2`. One connected with public and private networks and another connected with public network only.

Comment (SVilgelm) [inline cloudferrylib/os/network/neutron.py]: Actual result will be:

    public_notShared_network_t1 <=> router_t1 <=> private_notShared_network_t1
    public_notShared_network_t2 <=> router_t2 <=> private_notShared_network_t2

I will have only 2 routers.

Comment (SVilgelm) [inline cloudferrylib/os/network/neutron.py]: Bu I'll will check twice

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:813: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1811/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:813: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1838/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:814: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:814: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:814: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
$PULL_STATUS
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1841/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by japaniel** [comment]
+1 tested in staging

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**